### PR TITLE
Update Turkish Language / Board SD Printing: Show Actual Z Coordinate as Layer - Watch pause and cancel action - Bypass blocking operations on print cancel / Fix M27 time parameter /M701-M702 load/unload filament support

### LINKED ISSUE #367: [BUG] Stuck after cancel the print
### Descriptionn

When print some files from the sd card connected to board, if I cancel it screen get stucked in screen that says "Loading..."

### Steps to reproduce

Print, select onboadsd, select the file, stop, confirm, here is the issue.

TFT 35 V2 connected to SKR V1.4 @ Marlin latest build

### LINKED ISSUE #368: [BUG] Printing shows as still resumes when printer halted.
If printer halt when printing, screen continues to show as printing is resuming. (Example situation: When MINTEMP protection triggered and printer halted.)

Expected behaviour: Screen should go back to main menu and say "Printer halted, reset the printer." or asks "Reset printer?" (It connected with reset pin to board so this is possible.)

### LINKED ISSUE #834: [FR] Filament change buttons
To make the touch setup complete i would suggest to add the filament change option like under the normal marlin setup so that people with bowden setups can load and unload with one click. 

As an alternative would it be able to mod the aktual load and unload funktion so that instead of just 10mm at the time it feeds 400 mm and more ? I can't find these line in the code...

### LINKED ISSUE #1135: [BUG] On-screen Layer info not working printing from onboard SD
Hi.

When using the .26 firmware (latest build) and the build from the Biqu B1 repo the on-screen layer information is not working when printing from the onboard SD.

However if I print from any media connected to the TFT it works as expected.

My screen is a Biqu custom TFT35 e3 V3.

Thanks.

### PR DESCRIPTION
**Pull request by mehmetsutas**
* Update outdated Turkish translation

* During SD printing status screen showed 0mm as layer throughout the print. Actual Z position is shown which is more informative than 0mm

* Pause, paused and cancel keywords are added to the hostactions to watch the boardsd prints. Now when prints are canceled by host pc, it can be detected and LCD reverts to the statusmenu from printingmenu. Cancel action on printabort will also be added to Marlin.

* M524 processing was deferred by blocking operations such as extruder and bed heating processes. Break gcodes added before M524 in order to break the blocking operations and speed up the print cancel when printing from boardsd.

* Miliseconds were passed as parameter to M27 gcode which expects seconds as parameter. Fixed.

* Add support for M701-M702 Marlin load/unload filament gcodes. A new configuration option added for M701/702 support. A subpage added to the preaheat menu. Preheat menu moved to the main page in place of extrude menu. Extrude menu moved under move menu. After selecting the preheat option printer starts to heat on the next newly added page either filament load or unload option can be chosen or can exit without filament load/unload. Accordingly M701 or M702 gcode is sent to Marlin and Marlin handles the filament load/unload according to its setup on configuration_adv.h

resolves #1135
resolves #834
resolves #367
resolves #368

### CODE REVIEW

**Comment by oldman4U** [comment]
You are a gift from heaven!!!

Thank you👍🏻

**Review by traffic-light — CHANGES_REQUESTED** [review]
I scrolled trough your code. See the individual comments.

In short:
* Use curly braces, even when it's one line
* Place curly braces on its own line. Example:
````
}
Else
{
````

* Create a space after if, while, for etc. : `if (what == true)`
* Line up you're code with the rest of it

Thank you

**Comment by traffic-light** [comment]
It seems something went wrong with my suggestions. I hope it's clear enough

**Comment by traffic-light** [comment]
It seems something went wrong with my suggestions. I hope it's clear enough

**Comment by Maker-M** [comment]
Excellent news ! Forgive my ignorance, but what should we do? Is there a file to copy directly or a patch? Do I have to replace each red line with the corresponding green line by recompiling the affected files?

**Comment by oldman4U** [comment]
Hi Maker-M.

Are you mehmetsutas?

**Comment by mehmetsutas** [comment]
@traffic-light I will check your suggestions.

@Maker-M You may copy paste the changes for now. But you should better wait for the changes to be merged into the source. Meanwhile people may have different comments and point out some problems with the change if there are. After merge you can download the firmware again, compile and use it.

@oldman4U he is not. :)

**Comment by Maker-M** [comment]
> @ feu de circulation Je vérifierai vos suggestions.
> 
> @ Maker-M Vous pouvez copier-coller les modifications pour le moment. Mais vous devriez mieux attendre que les modifications soient fusionnées dans la source. Pendant ce temps, les gens peuvent avoir des commentaires différents et signaler certains problèmes avec le changement, le cas échéant. Après la fusion, vous pouvez télécharger à nouveau le micrologiciel, le compiler et l'utiliser.
> 
> @ oldman4U il ne l'est pas. :)

Great ! Thanks for your answers. 
Sorry oldman4U, i'm not...

**Comment by oldman4U** [comment]
Ahhhh, it is compiling;-)))

We had cases where users have been logging in using different devices which resulted in different names - so this is why I asked.

@mehmetsutas Thank you very much for contributing. Feel free to solve more long lasting issues or adding great features;-)!!!

**Comment by blueeagle69** [comment]
You are a genius. 
Layer info finally working when printing from onboard media.

I have noticed though that if no ABL is active (no mesh stored) the ABL icon doesn't show.
If I go to Marlin mode and run ABL then return to TFT the ABL icon appears. 

I don't think this is related,  as I'm sure this was mentioned elsewhere. 
But thought I'd mention it.
Great work on getting the layer info working!

**Comment by mehmetsutas** [comment]
@oldman4U it is easier to solve when the coder personally encounter the problem. But I will have look at the list.

I have added M701/702 load/unload gcode support. Will work on M600 and power loss recovery from boardsd.

**Comment by oldman4U** [comment]
Hi.

The problem is, that most of the developers in the past added a feature and then they disappeared. Most likely because of a completely chaotic handling of this repository. Some users including me, try to get the things organised again, by adding precompiles firmware which works and which are not several months old, by updating the ReadMe and removing things which are not working anymore. For me the most critical issue is the lack of engagement from BTT, so we try to work around this as much as we can. Thats one of the reasons we made the FR and Bug list, otherwise we would have hundreds of tickets with a search function which only works in parts and important issues and features on page 2 and later - and who reads more than the first page. So we got reports covering the same issues and the same time developers silently working on the same things.

Long story short. It would be great if we could convince you to stay a bit longer, in case you are interested and in case you have a bit time to spare.

Thank you

**Comment by mehmetsutas** [comment]
Created a pull request on Marlin for reporting the cancelled SD print.

https://github.com/MarlinFirmware/Marlin/pull/20223

**Comment by mehmetsutas** [comment]
> Hi.
> 
> The problem is, that most of the developers in the past added a feature and then they disappeared. Most likely because of a completely chaotic handling of this repository. Some users including me, try to get the things organised again, by adding precompiles firmware which works and which are not several months old, by updating the ReadMe and removing things which are not working anymore. For me the most critical issue is the lack of engagement from BTT, so we try to work around this as much as we can. Thats one of the reasons we made the FR and Bug list, otherwise we would have hundreds of tickets with a search function which only works in parts and important issues and features on page 2 and later - and who reads more than the first page. So we got reports covering the same issues and the same time developers silently working on the same things.
> 
> Long story short. It would be great if we could convince you to stay a bit longer, in case you are interested and in case you have a bit time to spare.
> 
> Thank you

I understand and I will try to support. It is hard for me to solve the problems of the features that I don't use but I will surely share any fix or enhancement that I make.

**Comment by oldman4U** [comment]
Hi. 

Could you please add the following to the description of this PR. Thank you 

resolves #1135

**Comment by ETE-Design** [comment]
@mehmetsutas Reallyd great to se you here, and nice finally filament chance is added... Mabye your the guy who can look at #1264 and make it?

**Comment by ETE-Design** [comment]
@bigtreetech Seem's like it's time to merge a Lot again...

**Comment by oldman4U** [comment]
> @bigtreetech Seem's like it's time to merge a Lot again...

Let's pray and hope for the best. It is almost a bit like BTT does not want this to be a successful repository.
